sql/sql_optimizer.cc

```diff
@@ -10162,16 +10163,51 @@ static bool internal_remove_eq_conds(THD *thd, Item *cond,
 
       if (new_item == NULL)
       {
-        if (is_cond_or)
+        bool cond_replace = false;
+        bool cond_value_equivalent = false;
+        switch (item->type())
+        {
+        case Item::Type::INT_ITEM:
         {
-          Item* cond_true =
-            new (thd->mem_root) Item_int((longlong)1, 1); // Always true
-          (void)li.replace(cond_true);
-          should_fix_fields = true;
+          Item_int* item_int = dynamic_cast<Item_int*>(item);
+          if (item_int)
+          {
+            cond_replace = true;
+						cond_value_equivalent = item_int->value ? true : false;
```

Comment:
1:  Code alignment. Code format is an important thing.
2: Makes your code more consice.